### PR TITLE
chapter 6: Modelling User

### PR DESCRIPTION
**Pull request by ngyihkai**
Ready

### CODE REVIEW

**Comment by ngyihkai** [comment]
Ready

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+
+  validates :name, presence: true
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}
```
Comment (ghost) [inline app/models/user.rb]:
```suggestion
  validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}, uniqueness: true
```
Why you have to make 2 validate email, when you need one

File: test/models/user_test.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (ghost) [inline test/models/user_test.rb]: You don't need to do test
delete this file please

**Review by ghost — COMMENTED** [review]
cmt

File: app/models/user.rb (hunk 1 above)
Comment (ngyihkai) [reply app/models/user.rb]: ar, just know it. fixed

**Comment by ngyihkai** [comment]
Ready

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,2 +1,13 @@
 class ApplicationController < ActionController::Base
+  before_action :switch_locale
+
+  private
+  def switch_locale(&action)
```
Comment (thaodtd-0665) [inline app/controllers/application_controller.rb]: no block using in method, you can remove it
```suggestion
  def switch_locale
```

**Comment by ngyihkai** [comment]
Ready

**Review by thaodtd-0665 — APPROVED** [review]

**Comment by thaodtd-0665** [comment]
merged

**Pull request merged**